src/taipy/core/pipeline/_pipeline_manager.py

```diff
@@ -164,13 +236,109 @@ def _submit(
     def __get_status_notifier_callbacks(pipeline: Pipeline) -> List:
         return [partial(c.callback, *c.params, pipeline) for c in pipeline.subscribers]
 
+    @classmethod
+    def _delete_all(cls):
+        """
+        Deletes all Pipelines.
+        """
+        scenarios = _ScenarioManagerFactory._build_manager()._get_all()
+        for scenario in scenarios:
+            scenario.pipelines = {}
+        if hasattr(cls, "_EVENT_ENTITY_TYPE"):
+            _publish_event(cls._EVENT_ENTITY_TYPE, "all", EventOperation.DELETION, None)
+
+    @classmethod
+    def _delete_many(cls, pipeline_ids: Iterable[str]):
+        """
+        Deletes Pipeline entities by a list of Pipeline ids.
+        """
+        scenario_manager = _ScenarioManagerFactory._build_manager()
+
+        scenario_ids_and_pipeline_names_map: Dict[str, List[str]] = {}
+        for pipeline in pipeline_ids:
+            pipeline_id = pipeline.id if isinstance(pipeline, Pipeline) else pipeline
+            pipeline_name, scenario_id = cls._breakdown_pipeline_id(pipeline_id)
+            pipelines_names = scenario_ids_and_pipeline_names_map.get(scenario_id, [])
+            pipelines_names.append(pipeline_name)
+            scenario_ids_and_pipeline_names_map[scenario_id] = pipelines_names
+
+        try:
+            for scenario_id, pipeline_names in scenario_ids_and_pipeline_names_map.items():
+                scenario = scenario_manager._get(scenario_id)
+                for pipeline_name in pipeline_names:
+                    del scenario._pipelines[pipeline_name]
+                scenario_manager._set(scenario)
+
+            if hasattr(cls, "_EVENT_ENTITY_TYPE"):
+                for pipeline_id in pipeline_ids:
+                    _publish_event(cls._EVENT_ENTITY_TYPE, pipeline_id, EventOperation.DELETION, None)
+        except (ModelNotFound, KeyError):
+            cls.__log_error_entity_not_found(pipeline_id)
+            raise ModelNotFound(cls._dir_name, pipeline_id)
+
+    @classmethod
+    def _delete_by_version(cls, version_number: str):
+        """
+        Deletes Pipelines by version number.
+        """
+        for scenario in _ScenarioManagerFactory()._build_manager()._repository._search("version", version_number):
+            cls._delete_many(scenario.pipelines.values())
+
+    @classmethod
+    def _delete(cls, pipeline_id: PipelineId):
+        """
+        Deletes a Pipeline by id.
+        """
+        pipeline_name, scenario_id = cls._breakdown_pipeline_id(pipeline_id)
+
+        if scenario := _ScenarioManagerFactory._build_manager()._get(scenario_id):
+            if pipeline_name in scenario._pipelines.keys():
+                scenario.remove_pipelines([pipeline_name])
+                if hasattr(cls, "_EVENT_ENTITY_TYPE"):
+                    _publish_event(cls._EVENT_ENTITY_TYPE, pipeline_id, EventOperation.DELETION, None)
+                return
+        raise ModelNotFound(cls._dir_name, pipeline_id)
```

Comment:
I think the way we raise `ModelNotFound` in SQLRepository is different, without the dir name

Reply:
just checked, it's `ModelNotFound(str(self.model_type.__name__), entity_id)` I think the end result would still be the same or closely the same. I'll have a look to see if we can stay consistent or not